src/main/java/org/opensearch/security/action/apitokens/ApiToken.java

```diff
@@ -28,37 +36,50 @@ public class ApiToken implements ToXContent {
     public static final String INDEX_PERMISSIONS_FIELD = "index_permissions";
     public static final String INDEX_PATTERN_FIELD = "index_pattern";
     public static final String ALLOWED_ACTIONS_FIELD = "allowed_actions";
+    public static final String EXPIRATION_FIELD = "expiration";
 
     private String name;
-    private final String jti;
+    private String jti;
     private final Instant creationTime;
     private List<String> clusterPermissions;
     private List<IndexPermission> indexPermissions;
+    private final long expiration;
 
-    public ApiToken(String name, String jti, List<String> clusterPermissions, List<IndexPermission> indexPermissions) {
+    public ApiToken(String name, List<String> clusterPermissions, List<IndexPermission> indexPermissions, Long expiration) {
         this.creationTime = Instant.now();
         this.name = name;
-        this.jti = jti;
         this.clusterPermissions = clusterPermissions;
         this.indexPermissions = indexPermissions;
+        this.expiration = expiration;
+    }
 
+    public ApiToken(String name, List<String> clusterPermissions, List<IndexPermission> indexPermissions) {
+        this.creationTime = Instant.now();
+        this.name = name;
+        this.clusterPermissions = clusterPermissions;
+        this.indexPermissions = indexPermissions;
+        this.expiration = Long.MAX_VALUE;
     }
 
     public ApiToken(
-        String description,
-        String jti,
+        String name,
         List<String> clusterPermissions,
         List<IndexPermission> indexPermissions,
-        Instant creationTime
+        Instant creationTime,
+        Long expiration
     ) {
-        this.name = description;
-        this.jti = jti;
+        this.name = name;
         this.clusterPermissions = clusterPermissions;
         this.indexPermissions = indexPermissions;
         this.creationTime = creationTime;
+        this.expiration = expiration;
 
     }
 
+    public void setJti(String jti) {
```

Comment:
Changed all fields to be final, and changed the call signature so that we get the JTI from the securityTokenManager first in order to create the immutable ApiToken.